pkg/controllers/user/helm/controller.go

```diff
@@ -134,9 +165,16 @@ func (l *Lifecycle) Updated(obj *v3.App) (runtime.Object, error) {
 		v3.AppConditionForceUpgrade.True(obj)
 	}
 	ns, err := l.NsClient.Get(obj.Spec.TargetNamespace, metav1.GetOptions{})
-	if err != nil && !errors.IsNotFound(err) {
-		return result, err
-	} else if errors.IsNotFound(err) {
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return result, err
+		}
+		if obj.Spec.MultiClusterAppName != "" {
```

Comment:
We are checking for this on line 136 and obj is not updated before this, so can we return this error earlier?

Reply:
This is just to ensure we return error only for app belonging to any mcApp